bundle/edu.gemini.spdb.shell/src/main/scala/edu/gemini/spdb/shell/misc/EphemerisPurgeCommand.scala

```diff
@@ -0,0 +1,52 @@
+package edu.gemini.spdb.shell.misc
+
+import edu.gemini.pot.sp.{ISPObservation, ISPProgram}
+import edu.gemini.spModel.rich.pot.sp.obsWrapper
+import edu.gemini.spModel.target.EphemerisPurge
+
+import scala.collection.JavaConverters._
+import scalaz._, Scalaz._
+
+/** Shell command for purging ephemeris data. */
+object EphemerisPurgeCommand {
+  sealed trait PurgeOption {
+    def include(o: ISPObservation): Boolean = this match {
+      case All          => true
+      case ObservedOnly => o.isObserved
+    }
+
+    def displayValue: String = this match {
+      case All          => "all"
+      case ObservedOnly => "observed"
+    }
+  }
+
+  object PurgeOption {
+    val AllValues = List(All, ObservedOnly)
+
+    def fromDisplayValue(s: String): Option[PurgeOption] =
+      AllValues.find(_.displayValue === s)
+
+    def usageString: String =
+      AllValues.map(_.displayValue).mkString("[", " | ", "]")
+  }
+
+  /** Option to purge ephemeris data from all observations in the program. */
+  case object All extends PurgeOption
```

Comment:
I think either way is fine.  I guess I like doing it that way when the implementations are short so you can easily see them all at once.  It keeps the definitions of the `case object`s short too so it's easier to see them all as well.